modules/standard/RecordParser.chpl

```diff
@@ -17,45 +17,136 @@
  * limitations under the License.
  */
 
-use IO, Regexp;
 
-/* A general purpose record reader/parser for channels.
+/*
+
+A general purpose record reader/parser for channels. Uses a regular expression
+to capture portions of the input, and then assigns each capture to each
+record field.
+
+This module makes the following assumptions:
+
+1. The file has been opened for reading and a reader channel has been created
+
+2. The programmer has created a record (NOT a class) that can have an arbitrary
+   number of fields that can be cast from strings (i.e no subrecords, arrays
+   etc).
+
+3. The number of captures in the regex string provided, MUST match the number
+   of fields in the record.
+
+4. The order in which the fields in the record appear must be in the same order
+   as the record being parsed.
+
+Example 1
+---------
+
+.. code-block:: chapel
+
+  use RecordParser;
+
+  record Bar {
+    var beer: string;
+    var Name: string;
+  }
+
+  var f = open("input1.txt", iomode.rw);
+  var fr = f.reader();
+
+  var M = new RecordReader(Bar, fr);
 
-   The assumptions on the programmer are:
-   (1) The programmer has opened the file in the proper mode 
+  var c = M.get();
+  writeln("The first record read is ", c);
 
-   (2) The programmer has created a record (NOT a class) that can have an arbitrary
-   number of fields that can be cast from strings (i.e no subrecords, arrays etc).
+  writeln("The rest of the records are:");
+  for r in M.stream() do
+    writeln(r);
 
-   (3) The number of captures in the regex string provided, MUST match the number of
-   fields in the record.
 
-   (4) The order in which the fields in the record appear must be in the same order
-   as the record being parsed. See test.chpl for an example of this.
+Example 2
+---------
+
+.. code-block:: chapel
+
+  use RecordParser;
+
+  record Beer {
+    var name:  string;
+    var beerId: int;
+    var brewerId: int;
+    var ABV: real;
+    var style: string;
+    var appearance: real;
+    var aroma: real;
+    var palate: real;
+    var taste: real;
+    var overall: real;
+    var time:  int;
+    var profileName:  string;
+    var text: string;
+  }
+  
+      
+  var strt = "\\s*beer/name: (.*)\\s*beer/beerId: (.*)\\s*beer/brewerId: (.*)\\s*beer/ABV: (.*)\\s*beer/style: (.*)\\s*review/appearance: (.*)\\s*review/aroma: (.*)\\s*review/palate: (.*)\\s*review/taste: (.*)\\s*review/overall: (.*)\\s*review/time: (.*)\\s*review/profileName: (.*)\\s*review/text: (.*)";
+
+  var N = new RecordReader(Beer, ffr, strt);
+  writeln("========== test of stream() ==============");
+  for r in N.stream() do
+    writeln(r);
```

Comment:
Similar to my comment for Curl and HDFS, I'd like to see another section header here, so that one could jump to the code definitions if one so chose.